app/src/main/java/org/oppia/android/app/home/HomeViewModel.kt

```diff
@@ -0,0 +1,178 @@
+package org.oppia.android.app.home
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import org.oppia.android.R
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.home.promotedlist.PromotedStoryListViewModel
+import org.oppia.android.app.home.promotedlist.PromotedStoryViewModel
+import org.oppia.android.app.home.topiclist.AllTopicsViewModel
+import org.oppia.android.app.home.topiclist.TopicSummaryClickListener
+import org.oppia.android.app.home.topiclist.TopicSummaryViewModel
+import org.oppia.android.app.model.OngoingStoryList
+import org.oppia.android.app.model.Profile
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.TopicList
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.domain.profile.ProfileManagementController
+import org.oppia.android.domain.topic.TopicListController
+import org.oppia.android.util.data.DataProvider
+import org.oppia.android.util.data.DataProviders.Companion.combineWith
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import org.oppia.android.util.logging.ConsoleLogger
+import org.oppia.android.util.parser.StoryHtmlParserEntityType
+import org.oppia.android.util.parser.TopicHtmlParserEntityType
+import org.oppia.android.util.system.OppiaClock
+
+private const val PROFILE_AND_ONGOING_STORY_COMBINED_PROVIDER_ID = "profile+ongoingStoryList"
+private const val HOME_FRAGMENT_COMBINED_PROVIDER_ID = "profile+ongoingStoryList+topicListProvider"
```

Comment:
Using `+` is correct format or should be `_`. WDYT @BenHenning

Reply:
I actually suggested this if I recall correct. We have few examples of combine elsewhere and the current examples aren't very clear. I think this is better since it's clearly indicating what providers are being combined.

Reply:
Would something like `profile_and_ongoingStoryList` be better? It does look like other IDs us `_` instead of `+`

Reply:
@jcqli Actually Ben is suggesting that we should follow this approach at other places too in the app. Therefore resolving this comment as you have done the right thing.